src/_pytest/debugging.py

```diff
@@ -89,156 +90,161 @@ def _is_capturing(cls, capman):
         return False
 
     @classmethod
-    def _import_pdb_cls(cls):
+    def _import_pdb_cls(cls, capman):
         if not cls._config:
             # Happens when using pytest.set_trace outside of a test.
             return pdb.Pdb
 
         pdb_cls = cls._config.getvalue("usepdb_cls")
+        if cls._wrapped_pdb_cls and cls._wrapped_pdb_cls[0] == pdb_cls:
+            return cls._wrapped_pdb_cls[1]
+
         if not pdb_cls:
-            return pdb.Pdb
+            pdb_cls = pdb.Pdb
+        else:
+            modname, classname = pdb_cls
+
+            try:
+                __import__(modname)
+                mod = sys.modules[modname]
+
+                # Handle --pdbcls=pdb:pdb.Pdb (useful e.g. with pdbpp).
+                parts = classname.split(".")
+                pdb_cls = getattr(mod, parts[0])
+                for part in parts[1:]:
+                    pdb_cls = getattr(pdb_cls, part)
+            except Exception as exc:
+                value = ":".join((modname, classname))
+                raise UsageError(
+                    "--pdbcls: could not import {!r}: {}".format(value, exc)
+                )
 
-        modname, classname = pdb_cls
+        import _pytest.config
 
-        try:
-            __import__(modname)
-            mod = sys.modules[modname]
+        class PytestPdbWrapper(pdb_cls, object):
```

Comment:
Can we move this class creation to a module level function? This class in the middle of the code like this makes it harder to understand than it could be.

Reply:
Will move it into a classmethod.